bundles/org.eclipse.passage.lic.base/src/org/eclipse/passage/lic/internal/base/conditions/MatchingRuleForIdentifier.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2020 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lic.internal.base.conditions;
+
+import java.util.function.Supplier;
+
+import org.eclipse.passage.lic.internal.api.conditions.MatchingRule;
+
+@SuppressWarnings("restriction")
+public final class MatchingRuleForIdentifier implements Supplier<MatchingRule> {
+	private final String identifier;
+
+	public MatchingRuleForIdentifier(String identifier) {
+		this.identifier = identifier;
+	}
+
+	@Override
+	public MatchingRule get() {
+		switch (identifier.toLowerCase()) {
+		case "compatible": { //$NON-NLS-1$
```

Comment:
May be we can escape from duplicating literals and return the matching rule by its identifier, as
`new MatchingRuleCompatible().identifier()=="compatible"`

Reply:
Do not like this either, but singletones are worse. but see fine solution at this level. Will revise this when application code is reworked: we probably will be able to construct sort of a storage map one for a, say, license parsing session.